src/components/sections/self-prfile-edit/index.tsx

```diff
@@ -297,16 +318,125 @@ const SelfProfileEditSection: FC<SelfProfileEditSectionProps> = ({ profileData,
     }
+  
   };
-const onVerificationSuccess = () => {
-  setShowOtpVerification(false);
-  setShowCongratulationModal(true);
-  setVerificationData(null);
-}
-  // const verifyOtpHandler = async () => {
-  //   const data = await verifyVerificationCode({verificationId: verificationData?.verificationId || '', otp: otp}).unwrap();
-  // }
+
+ const [deleteAccount, {isLoading: isDeleting}] = selfProfileApi.useDeleteAccountMutation();
+  const [showDeleteAccountModal, setShowDeleteAccountModal] = useState(false);
+  const [showDeleteReasons, setShowDeleteReasons] = useState(false);
+  const [selectedDeleteReason, setSelectedDeleteReason] = useState<string[]>([]);
+  const { data: deleteAccountReasons } = selfProfileApi.useGetDeleteAccountReasonsQuery();
+  const deleteReasons = useMemo(() => {
+    return deleteAccountReasons?.data.map((reason) => ({
+      label: reason.reason,
+      value: reason._id,
+    }));
+  }, [deleteAccountReasons]);
+
+  const [deleteError, setDeleteError] = useState('');
+  const [deleteOtherReason, setDeleteOtherReason] = useState('');
+
+  const handleDeleteReasonChange = (selectedValues: string[]) => {
+    setSelectedDeleteReason(selectedValues);
+  };
+  const deleteAccountHandler = () => {
+    setShowDeleteAccountModal(true);
+  };
+  const onDeleteModalClose = () => {
+    setShowDeleteAccountModal(false);
+    setShowDeleteReasons(false);
+    setSelectedDeleteReason([]);
+    setDeleteOtherReason('');
+    setDeleteError('');
+  };
+  const onDeleteConfirm = () => {
+    setShowDeleteReasons(true);
+  };
+  const onDeleteCancel = () => {
+    setShowDeleteAccountModal(false);
+    onDeleteModalClose();
+  };
+
+    const handleDeleteReasonSubmit = async () => {
+      if (selectedDeleteReason.length == 0) {
+        setDeleteError('Please select at least one reason');
+        return;
+      }
+      if (selectedDeleteReason?.[0] == 'Other' && deleteOtherReason == '') {
+        setDeleteError('Please specify the reason');
+        return;
+      }
+      const reason =
+        selectedDeleteReason?.[0] == 'Other'
+          ? deleteOtherReason
+          : deleteReasons?.find((option) => option.value === selectedDeleteReason?.[0])?.label || '';
+      const deleteAccountPayload = {
+        id: profileData?._id || '',
+        reason: reason,
+        deleteConfirmation: 'false',
+      };
+      try {
+        const data = await deleteAccount(deleteAccountPayload).unwrap();
+        showToast({
+          message: 'An email has been sent to your registered email address for account deletion confirmation.',
+          messageType: 'success',
+          duration: 5000,
+        });
+        onDeleteModalClose();
+      } catch (error) {
+        console.log(error, 'delete account error');
+        const errorData = error as { data: { message: string } };
+        showToast({
+          message: errorData?.data?.message || 'Something went wrong please try after sometime',
```

Comment:
Do not hardcode strings directly in component. you can create seperate for error